tests/commands/test_compile.py

```diff
@@ -144,3 +144,40 @@ def test__compile_auto_account_flag(mock_subprocess, contract_name, flag):
         stdout=mock_subprocess.PIPE,
     )
     mock_process.communicate.assert_called_once()
+
+
+@pytest.mark.parametrize(
+    "contract_name, directory, cairo_path, expected_internal_cairo_path",
+    [
+        ("Contract1", "contracts", None, "contracts"),
+        ("Contract3", "src", "contracts:src", "contracts:src"),
+        ("Contract1", None, "src", "src"),
+        ("Contract4", None, None, CONTRACTS_DIRECTORY),
+    ],
+)
+def test__compile_cairo_path(
+    mock_subprocess, contract_name, directory, cairo_path, expected_internal_cairo_path
+):
+    path = f"path/to/{contract_name}.cairo"
+
+    mock_process = Mock()
+    mock_subprocess.Popen.return_value = mock_process
+
+    # Use named arguments to allow changing the order without modifying tests
```

Comment:
what do you mean by changing the order?

Reply:
Order of the arguments in the signature of the function.

Reply:
i don't understand why would they change

Reply:
I'm not implying they should (or go to) change, but in the case for some reason is done (because it make more sense for any reason to reorganize the arguments), we don't need to look in the rest of the code for the method to update the order. 

It is similar to the pattern for initialization of structs in Solidity (named arguments allowing reorganization of the order of the members of the struct, for potential packaging in this case, but the principle is the same one).

Reply:
if it's a design decision i'd add that to contributing and not in here. i vouch for minimal comments when they're really helping understand the code it's being read.

Reply:
Makes sense, I removed the comment.